File: CoreJava/doug's-dog-doors-app/src/components/DogDoor.java
package components;

import java.util.ArrayList;
import java.util.Timer;
import java.util.TimerTask;

public class DogDoor {
	private boolean open;
	private ArrayList<Bark> allowedBarks=new ArrayList<Bark>();

	public DogDoor() {
		this.open = false;
	}

	public void open() {
		System.out.println("The Dog Door Open.");
		this.open = true;

		final Timer timer = new Timer();
		timer.schedule(new TimerTask() {
			@Override
			public void run() {
				close();
				timer.cancel();
			}
		}, 5000);
	}

	public void close() {
		System.out.println("The Dog Door Close.");
		this.open = false;
	}

	public boolean isOpen() {
		return open;
	}
	
	public void addAllowedBark(Bark bark){
		allowedBarks.add(bark);
	}
	
	public ArrayList<Bark> getAllowedBarks(){
		return allowedBarks;
	}

}

File: C#/OOP/DogDoorApp/DogDoorApp/DogDoor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Timers;


namespace DogDoorApp
{
    class DogDoor
    {
        private bool _open;
        private List<Bark> _allowedBarks;

        public DogDoor()
        {
            _allowedBarks = new List<Bark>();
            _open = false;
        }

        public void AddAllowedBark(Bark bark)
        {
            _allowedBarks.Add(bark);
        }

        public List<Bark> GetAllowedBark
        {
            get
            {
                return _allowedBarks;
            }
        }

        public bool isOpen()
        {
            return _open;
        }

        public void Close()
        {
            Console.WriteLine("The Dog Door Closes");
            _open = false;
        }

        public void Open()
        {
            Console.WriteLine("The Dog Door Opens");
            _open = true;

            Timer timer = new Timer { Interval = 5000 };
            timer.Elapsed += OnTimedEvent;
        }

        void OnTimedEvent(Object source, ElapsedEventArgs e)
        {
            Close();
        }
    }
}
